CRM/Contribute/Form/UpdateSubscription.php

```diff
@@ -87,6 +87,7 @@ public function preProcess() {
       catch (CRM_Core_Exception $e) {
         CRM_Core_Error::statusBounce(ts('There is no valid processor for this subscription so it cannot be edited.'));
       }
+      $this->_paymentProcessorObj = $this->_paymentProcessor['object'];
```

Comment:
@monishdeb @eileenmcnaughton I've just run into this line on master and it seems to be breaking the "update subscription" functionality for me.  I'm using the smartdebit extension and I get a php crash because $this->_paymentProcessorObj is not an object.  Running through a debugger the "object" is actually $this->_paymentProcessor and there is no 'object' array element.  Do either of you understand what this line is supposed to do - is it something that needs updating in my smartdebit extension?

Reply:
arg -does this work?

https://github.com/eileenmcnaughton/civicrm-core/commit/3bc67dd2b7202ebf30491c4d72ec6956f9424872

The assigns to the tpl seem no longer required - we call the payment processor to determine what fields can be edited & what text should be rendered

    $this->editableScheduleFields = $this->_paymentProcessorObj->getEditableRecurringScheduleFields();

    $changeHelpText = $this->_paymentProcessorObj->getRecurringScheduleUpdateHelpText();

Reply:
Yes it does!  Well I tested UpdateSubscription at least and that seems to work fine with that additional patch